### ISSUE TITLE
options in advanced preferences are doubled

### ISSUE THREAD

**Issue by hermestrismi** [wesnoth/wesnoth#9581]
### Game and System Information

- **Version: BfW 1.19.5**
- **OS: Windows 10**

### Description of the bug

In the **advanced preferences**, the options are repeated twice
![20241113_090059_mfnr](https://github.com/user-attachments/assets/963d71bb-01c2-4c51-b7d8-3e1ddfd159b0)

### Steps to reproduce the behavior

1. open **preferences**
2. select **advanced**

### Expected behavior

Each option must be unique 

### Additional context

Didn't test the effect of omitting one option for instance but in the log there is an error about conflicts between **default** GUIs

**Comment by soliton-** [comment]
Did you include `data/advanced_preferences.cfg` twice? Maybe in a core?

**Comment by hermestrismi** [comment]
> Did you include `data/advanced_preferences.cfg` twice? Maybe in a core?

I am sure I didn't modify the file manually but I open wesnoth from two users interfaces (administrator + invited) which imply two different files of preferences, logs, ...

**Comment by soliton-** [comment]
> I am sure I didn't modify the file manually

I wouldn't know how you would manage to include a file twice by editing the file itself. There would be another file that includes the file twice.

Normally the file is included just here: https://github.com/wesnoth/wesnoth/blob/df3f2b2653fe198cd4f2e2662631f11812525a92/data/_main.cfg#L99
If it is included elsewhere (in some addon perhaps) as well then exactly what you report happens and that is perfectly normal.

I guess the GUI stuff you vaguely mentioned sounds like everything is included twice.

> but I open wesnoth from two users interfaces (administrator + invited) which imply two different files of preferences, logs, ...

Not sure how that relates. Does it only happen for one of the users?

**Comment by hermestrismi** [comment]
> > I am sure I didn't modify the file manually
> 
> I wouldn't know how you would manage to include a file twice by editing the file itself. There would be another file that includes the file twice.
> 
> Normally the file is included just here: https://github.com/wesnoth/wesnoth/blob/df3f2b2653fe198cd4f2e2662631f11812525a92/data/_main.cfg#L99
> If it is included elsewhere (in some addon perhaps) as well then exactly what you report happens and that is perfectly normal.
> 
> I guess the GUI stuff you vaguely mentioned sounds like everything is included twice.
> 
> > but I open wesnoth from two users interfaces (administrator + invited) which imply two different files of preferences, logs, ...
> 
> Not sure how that relates. Does it only happen for one of the users?
For the first point, no. I don't have such file or macro in my addons. 
For the second part, I quite the computer so I can't test that however I have also multiple Wesnoth versions (from 1.0 to 1.19). is such bug can be caused by this?

**Comment by Wedge009** [comment]
I suspect language might be an impediment here. I can't replicate the issue and it sounds like soliton hasn't either. Are you saying this only happens with two Wesnoth application instances?

@Pentarctagon changed some preferences code in #9151 which is included from 1.19.3. Not sure if that is necessarily related, but it might help if you can test 1.19.2 and 1.19.3 for this.

**Comment by Pentarctagon** [comment]
I haven't seen this happen before either, for what it's worth.

**Comment by soliton-** [comment]
> I suspect language might be an impediment here. I can't replicate the issue and it sounds like soliton hasn't either.

I can replicate it in the way I mentioned. Include `advanced_preferences.cfg` another time in an addon for example and you'll get every advanced preference twice. There isn't much wrong with that though. Just don't do that if you don't want every option twice.

I bet I could also produce some error about duplicate GUI themes if I include GUI stuff twice.

**Comment by hermestrismi** [comment]
After checking, I found two **preferences** files. one in the main file of the documents/my games/wesnoth 1.19, the other in the /sync file.
However, I don't have any clue about how that happened. 
Is it possible that when I deleted the previous version and replaced it with the actual version that the previous files appended ?

**Comment by Pentarctagon** [comment]
No, that's intentional. With 1.20 on Steam everything in the `sync` folder (including that part of the preferences) will be synced with Steam Cloud. Any preferences that only exist in the preferences file outside of the `sync` folder will not be synced via Steam Cloud.

**Comment by Wedge009** [comment]
Yes, that was from a previous change and only one preferences file is used at a time. Checking the file paths in the About dialogue should tell you which one is being used.

**Comment by gfgtdf** [comment]
@hermestrismi it would be nice to know when exactly it hppens, like, only during a  game, or only after reloading? the screenshot in the first post shows a game in the background but the steps seems to suggests that all you need to do is to open wensoth and then open the prefences dialog.

**Comment by Wedge009** [comment]
That's why neither Pent nor I could replicate it. soliton seemed to replicate it via an add-on.

**Comment by eyalroz** [comment]
Also seeing this, after:

1. Building and running 1.19.7
2. Importing settings from 1.17
3. Installing Orcish Incursion, then Asheviere's Shadow
4. Starting the first scenario of Orcish Incursion

**Comment by soliton-** [comment]
Looks like it happens each time you press F5 on the title screen or install/remove an addon.
